mage/main.go

```diff
@@ -137,6 +138,15 @@ func Parse(stdout io.Writer, args []string) (inv Invocation, mageInit, showVersi
 func Invoke(inv Invocation) int {
 	log := log.New(inv.Stderr, "", 0)
 
+	// If verbose is still false, we're going to peek at the environment variable to see if
+	// MAGE_VERBOSE has been set. If so, we're going to use it for the value of MAGE_VERBOSE.
+	if inv.Verbose == false {
```

Comment:
this feels like it belongs in Parse().  Parse is where we take environmental factors (like os.Args and os.Environ) and convert them into function arguments (in our case, an Invocation struct)

Reply:
Ideally I'd like ParseAndRun to take a []string{} of the same type os.Environ returns, and then pass that to Parse.  Then Main() can just pass os.Environ() ... and it makes testing easy without worring about environment variables on the test machine.

Reply:
And then you could just test Parse and not need to worry about actually calling Invoke, since that would just be testing the Verbose field in Invocation, which should be separate.

Reply:
If this sounds like too much work, feel free to just pop this into parse, and I'll do the rest :)

Reply:
Originally I had it in parse.Test was failing because the Test was calling 'Invoke()' directly. if it's in Parse, Invoke will never see it.

Reply:
Picking up my kids. Will peek at this again when I get back.